application/app/controllers/dataverse/datasets_controller.rb

```diff
@@ -12,18 +12,23 @@ def show
 
   def download
     file_ids = params[:file_ids]
-    project = @service.initialize_project(@dataset)
-    unless project.save
-      flash[:alert] = "Error generating project: #{project.errors}"
-      redirect_to downloads_path
-      return
+    project = Project.find(params[:active_project])
+    if project.nil?
+      project = @service.initialize_project(@dataset)
+      unless project.save
+        redirect_back fallback_location: root_path, alert: "Error generating project: #{project.errors}"
+        return
+      end
     end
```

Comment:
Isn't active project stored as a user setting? why should we need to pass it as a param?

It this is stored as a model class, a good refactor for a future PR could be
```ruby
project = Project.find_or_initialize_active_project
```
By having this in the model level, in the future we can handle the active project at the model level when we run the `destroy` method

Reply:
fixed